### ISSUE TITLE
Using `info` or `config` in a `multi/exec`

### ISSUE THREAD

**Issue by chrisskilton** [luin/ioredis#411]
I have just migrated from `node_redis`. I had a check script that did the following: 

```javascript
redis.multi()
        .info()
        .config('get', 'maxmemory')
        .exec()
        .then(console.log) //works in node_redis, outputs [] in ioredis
```

Running the commands individually works (not using `multi`). Is this by design? I only used the multi  in the first place to reduce a bit of nesting as I need to munge the results.

**Comment by luin** [comment]
It should return with the correct result:
![image](https://cloud.githubusercontent.com/assets/635902/21211960/2b623d8a-c2c3-11e6-9941-2e881744e367.png)

Could you please provide a minimal reproducible example?

**Comment by chrisskilton** [comment]
I had the code wrapped in `redis.on('connect')`. Here's a reproducible example.

script:
```javascript
'use strict';

const Redis = require('ioredis');
const redis = new Redis();

redis.on('connect', () => {
    console.log('redis connected');
    redis.info().then(info => {
        console.log('redis.info response', info);
    });

    redis.config('get', 'maxmemory').then(maxMem => {
        console.log('redis.config response', maxMem);
    });

    redis.multi().info().config('get', 'maxmemory').exec().then(results => {
        console.log('redis.multi response', results);
    });
});
```

```
$ node check-scripts/test.js
redis connected
redis.info response # Server
redis_version:3.0.7
redis_git_sha1:00000000
redis_git_dirty:0

...

redis.config response [ 'maxmemory', '3145728000' ]
redis.multi response []
```

The multi _does_ work outside the `connect` callback.

**Comment by luin** [comment]
Thank for the report! I just confirmed this is a bug due to a race condition. It happens when sending pipeline/transaction that includes `info` or `config` command on the `"connect"` event.

I'm going to fix this today. BTW, It's recommended to handle the `"ready"` event instead of the `"connect"` event, since the latter one is a low level event emitted by the "net" module directly.

**Comment by luin** [comment]
Fixed in v2.4.3. 🍻

**Comment by chrisskilton** [comment]
closing this as it's fixed
